test/fixtures/dialects/bigquery/select_identifiers.sql

```diff
@@ -0,0 +1,19 @@
+--Identifiers can start with an underscore in BigQuery
+-- and can contin just _ and 0-9
+SELECT _01
+FROM _2010_01;
+
+--Identifiers can start with an underscore in BigQuery
+-- and can contin just _ and 0-9
+SELECT col_a AS _
+FROM table1;
+
+-- TODO: Currently we don't support this but should
+-- Table names can contain dashes from FROM and TABLE clauses
+-- But reluctant to add to general naked_identifier grammar and not
+-- sure worth adding specific syntax for this unless someone requests it
```

Comment:
Hey, original poster of the linked issue here.  

We've learned to escape dashes to deal with lack of SQLFluff support but I did want to note that we do use dashes in our project names and so native support for it would be fantastic.  Don't know how common this usage pattern is but just wanted to add a +1 here :)

Reply:
Well there's two options:
1. Add this to `NakedIdentifierSegment` and `NakedIdentifierSegmentFull`. But then dashes will also be allowed in unquoted column names and aliases. Which they are not allowed. So overly permissive. Not the worst in the world and if it's invalid identifier then BigQuery will pick it up when it's run, but ideally SQLFluff would have already picked this up.
2. Add a new segment (`NakedIdentifierSegmentWithDashes`?) and then slot that in in place of `NakedIdentifierSegment`, but that's a bigger change I wouldn't want to do in this PR. And could be tricky to identify all those cases.

Thoughts on preference?

Also BigQuery does allow dashes in quotes too, that SQLFluff does support that. But presume I can't persuade you to move to that style? 😁

Reply:
I'm not using BigQuery at the moment, but the first sounds reasonable for now. SQLFluff's dialects are generally supersets of the actual language (for various reasons). In general, my philosophy is that SQLFluff is a linter/fixer, **not** a 100% syntax checker.

Reply:
Normally would agree @barrywhart but these [dashes are very much the exception](https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical):

> Table name identifiers have additional syntax to support dashes (-) when referenced in FROM and TABLE clauses.

So were being quite liberal with ALL naked identifiers for one very specific (and I'd argue, as a BigQuery user niche) use case.

Let me mull over it some more in my head before merging this...

@kwd2126 would love to hear your thoughts?

Reply:
@tunetheweb, now that you've laid out the blockers here, I agree with your logic of not dealing with this case yet.  I definitely think solution #1 would be an net regression for our team -- as I said, we are used to quoting our project name to deal with SQLFluff issues but having proper unquoted column name parsing break would create way more problems than this would solve.  #2 would be ideal but as you said that's not an easy fix.

In the interest of contributing to this project outside of constantly raising issues, maybe it would be valuable to have documentation for differences between BigQuery and SQLFluff ideal syntax to capture these edge cases?  I'd be happy to go through the BigQuery documentation and start playing around with finding these edge cases if the maintainers think this would be valuable to the project's existing documentation.

Plus, I'd feel less worried about the bug (and any other differences between SQLFluff and BigQuery syntax in general) if there was some document that new employees could use to better understand these edge cases.

Reply:
I think we should document as GitHub Issues and work through them, rather than in a separate doc.

Could you raise one for this case? I'll add the BigQuery label (annoyingly only maintainers can add labels) and then it's easily searchable for known BigQuery issues: https://github.com/sqlfluff/sqlfluff/labels/bigquery

And feel free to raise whatever other ones you find. We like to know the issues! Just can't promise when/if we'll get round to fixing them!

Reply:
> but having proper unquoted column name parsing break would create way more problems than this would solve.

Just checking you understood the drawbacks of 1 here @kdw2126 ?

I'm saying this currently breaks:

```sql
select col-1 as alias-1
```

If we went with option 1 (which is the simpler change, and potentially could be included in this fix as same regex to change), this would no longer break SQLFluff and would be considered valid BigQuery SQL, even though it is not.

No extra _valid_ SQL would break as a result of that change. It would just allow some extra _invalid_ SQL to get through now, whereas before, it would have blocked it. As @barrywhart says SQLFluff is not a perfect match for each dialect and often is a bit more liberal than the actual dialect. I'm just not sure if this change would make it _too_ liberal.

Reply:
Sorry for the lack of clarity, edited my comment to clarify.  

I still believe that option 1 would not be desirable.  I understand that using SQLFluff to figure out if a command is valid BigQuery syntax in every case is not something the tool is designed to do, but I do appreciate that it is close enough to BigQuery syntax that it flags the example you gave with a parse error.  Having it miss invalid syntax examples like this would be a regression from my point of view, and the extra time we'd save by not having to quote our project view is trivial by comparison.

Understood on the documentation point, will continue to raise issues and look for other documentation issues to try to contribute.

Reply:
OK sounds good. Will merge this as is then.